tests/js/chartAuthoringTests.js

```diff
@@ -155,8 +212,93 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
-        jqUnit.assertDeepEq("Model is relayed between dataEntryPanel and pieChart", floe.tests.chartAuthoring.dataSet, that.chartAuthoringInterface.pieChart.model.dataSet);
+        var dataEntryPanelDataSet = that.chartAuthoringInterface.dataEntryPanel.model.dataSet;
+        var pieDataSet = that.chartAuthoringInterface.pieChart.model.dataSet;
+
+        // For looping pieDataSet in parallel
+        var indexCounter = 0;
+
+        fluid.each(dataEntryPanelDataSet, function (dataEntryPanelData, key) {
+
+            // Don't compare if the dataEntryPanelData has the default blank
+            // values (these don't get relayed)
+            if (dataEntryPanelData.value !== null && dataEntryPanelData !== "") {
+                jqUnit.assertEquals("Comparing relayed dataset - data value of pieChart data in position " + indexCounter + " matches", dataEntryPanelData.value, pieDataSet[indexCounter].value);
+                jqUnit.assertEquals("Comparing relayed dataset - label value of pieChart data in position " + indexCounter + " matches", dataEntryPanelData.label, pieDataSet[indexCounter].label);
+                jqUnit.assertEquals("Comparing relayed dataset - id value of pieChart data in position " + indexCounter + " matches expected key", key, pieDataSet[indexCounter].id);
+            }
+            indexCounter++;
+        });
+    };
+
+    // Verifies that a play button click triggers the sonification request to
+    // begin - there are items in the queue but currentlyPlayingData is empty
+    floe.tests.chartAuthoringTester.verifySonificationQueued = function (that) {
+        var sonifier = that.chartAuthoringInterface.sonifier;
+        jqUnit.assertEquals("Sonifier queue is full", 3, sonifier.model.sonificationQueue.length);
+        jqUnit.assertUndefined("No currentlyPlayingData", sonifier.currentlyPlayingData);
+    };
+
+    // Verify a stop button click triggers the sonification to stop -
+    // the queue is empty, no currentlyPlayingData
+    floe.tests.chartAuthoringTester.verifySonificationStopped = function (that) {
+        var sonifier = that.chartAuthoringInterface.sonifier;
+        jqUnit.assertEquals("Sonifier queue is empty", 0, sonifier.model.sonificationQueue.length);
+        jqUnit.assertUndefined("No currentlyPlayingData", sonifier.currentlyPlayingData);
+    };
+
+    // Verifies datapoint play behaviour in the context of the chart authoring
+    // tool
+    floe.tests.chartAuthoringTester.verifyDatapointPlay = function (that) {
+        jqUnit.assertDeepEq("currentlyPlayingData is properly relayed between chartAuthoring component and sonifier subcomponent", that.chartAuthoringInterface.sonifier.model.currentlyPlayingData, that.model.currentlyPlayingData);
+        jqUnit.assertDeepEq("activeRowId is the id of the currently playing data in the sonifier", that.chartAuthoringInterface.sonifier.model.currentlyPlayingData.id, that.chartAuthoringInterface.pieChart.legend.model.activeRowId);
```

Comment:
At line 281-284, can we also verify the actual ids, such as when the first data point is playing, the id is 0.